docs/blog/2019-03-27-security-concerns-unique-to-static-sites/index.md

```diff
@@ -0,0 +1,128 @@
+
+Why address this scenario? Two reasons:
+
+- **Simplicity** - Consider a small blog or non-profit site where authentication is not already needed. To develop it would require a lot of time and long term investment. For users authenticating to small blog just to hit "like" is bad user experience and is probably a hurtle they will not bother jumping.
+
+- **Privacy** - No matter the size of your platform, requiring user to authenticate means having to secure private data. This requires time and money. Alternatively, as a platform, you may decide your users' privacy is more important.
+
+Using Medium's "clap" feature as example, what are the issues? First, we need some kind of database to store the claps. This is easy enough, but do we leave no security mechanism on this database for reads and writes? We could add security, but then the API key we need has to be shipped in the code and the user can access it. Either way, your more nefarious users could access this database; reading and (more importantly) writing to their devious heart's content.
+
+There is a small blog that does this very thing. They build the Firebase credentials into the code at build time. Because this is a small blog, and their is no real incentive for someone to add or delete claps, this blogger is probably safe. What is the worse case scenario? Someone deletes their claps? Gives all their posts a million claps? Maybe this is defeating the point of a security post. But security is bout threat assessment. This blogger is clearly not worried about a nation state hacking their claps or someone gaming a system. Nothing but authentication is going to solve those kinds of issues.
+
+Assuming Russia is not trying to hack you and you would like to better secure your claps, what are you options? You need an API.
+
+That sounds hard? Go checkout Netlify, They will provide great static hosting and allow you to deploy serverless functions right along side your static site. They are not the only ones, Azure, Google, Amazon, Zeit.co, and others all have solutions.
+
+What does this API allow? First, you can secure your database. Any API keys can be secured on your serverless...server 😭. The point is, they will be secure. But what does this gain you? After all, instead of pillaging your database directly you have given them an API through which to pillage.
+
+Control, you have gained the ability to add rate limiting, better logging, auto blocking of IPs, etc. You can throw the proverbial "secure API handbook" (except for the authentication part) at your users to make sure they behave. More aggressively, you could use a secure cookie for your site to uniquely and anonymously identify each user. This would assist in the previously mentioned security practices and, while not hard to delete a cookie, adds another layer of security.
```

Comment:
```suggestion
**Control:** You have gained the ability to add rate limiting, better logging, auto blocking of IPs, etc. You can throw the proverbial "secure API handbook" (except for the authentication part) at your users to make sure they behave. More aggressively, you could use a secure cookie for your site to uniquely and anonymously identify each user. This would assist in the previously mentioned security practices and, while not hard for a user to delete a cookie, it adds another layer of security.
```